crates/opencascade/src/primitives.rs

```diff
@@ -807,7 +863,65 @@ impl AsRef<Shape> for Shape {
     }
 }
 
+impl From<Vertex> for Shape {
+    fn from(vertex: Vertex) -> Self {
+        let shape = ffi::cast_vertex_to_shape(&vertex.inner);
+        let inner = ffi::TopoDS_Shape_to_owned(shape);
+
+        Shape { inner }
+    }
+}
+
+impl From<Edge> for Shape {
+    fn from(edge: Edge) -> Self {
+        let shape = ffi::cast_edge_to_shape(&edge.inner);
+        let inner = ffi::TopoDS_Shape_to_owned(shape);
+
+        Shape { inner }
+    }
+}
+
+impl From<Wire> for Shape {
+    fn from(wire: Wire) -> Self {
+        let shape = ffi::cast_wire_to_shape(&wire.inner);
+        let inner = ffi::TopoDS_Shape_to_owned(shape);
+
+        Shape { inner }
+    }
+}
+
+impl From<Face> for Shape {
```

Comment:
I really like to see the those From implementations for all those primitives! However, many of the primitives also have a `to_shape` function, which could be deduplicated by using `.into()`. Even better may be to implement a trait which automatically provides `to_shape()` using the From implementations:

```rust
pub trait ToShape {
    fn to_shape(self) -> Shape;
}

impl<T: Into<Shape>> ToShape for T {
    fn to_shape(self) -> Shape {
        self.into()
    }
}
```

Reply:
Great suggestion! I felt like I still wanted to hang onto the `to_shape()` function because it seemed clearer than `.into()` with type ascription. I'll play around with your idea and get it added.